loader/loader.go

```diff
@@ -1188,110 +1189,308 @@ func fetchMatchedLiteral(ctx *tcontext.Context, router *router.Table, schema, ta
 	return targetSchema, targetTable
 }
 
-func (l *Loader) restoreData(ctx context.Context) error {
-	begin := time.Now()
+// restore schema job
+type restoreSchemaJob struct {
+	loader           *Loader
+	session          *DBConn
+	database         string
+	table            string // empty if it's a schema of database
+	pathOfSchemaFile string
+}
 
-	baseConn, err := l.toDB.GetBaseConn(ctx)
-	if err != nil {
-		return err
+// check is schema of database
+func (job *restoreSchemaJob) isDB() (yes bool) {
+	yes = (job.table == "")
+	return
+}
+
+// check is schema of table
+func (job *restoreSchemaJob) isTable() (yes bool) {
+	yes = !job.isDB()
+	return
+}
+
+// `jobQueue` of schema restoring which support consumptions concurrently
+type jobQueue struct {
+	ctx    context.Context
+	wg     sync.WaitGroup         // wait group of jobs
+	msgq   chan *restoreSchemaJob // job message queue channel
+	closed bool                   // msgq closed flag
```

Comment:
not necessary to use a `close` flag 🤔

Reply:
It's necessary to provide a status to support inside functional methods and present to the outside program.

Reply:
IMO this is a private structure, the responsibility is put more on caller

Reply:
It's true, but `closed` do provide support for inside functional implementation, it's not just a presentation.